features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListPresenter.kt

```diff
@@ -129,10 +138,23 @@ class RoomListPresenter @Inject constructor(
                         roomId = event.roomListRoomSummary.roomId,
                         roomName = event.roomListRoomSummary.name,
                         isDm = event.roomListRoomSummary.isDm,
+                        markAsUnreadFeatureFlagEnabled = markAsUnreadFeatureFlagEnabled == true,
+                        hasNewContent = event.roomListRoomSummary.hasNewContent
                     )
                 }
                 is RoomListEvents.HideContextMenu -> contextMenu = RoomListState.ContextMenu.Hidden
                 is RoomListEvents.LeaveRoom -> leaveRoomState.eventSink(LeaveRoomEvent.ShowConfirmation(event.roomId))
+                is RoomListEvents.MarkAsRead -> coroutineScope.launch {
+                    val receiptType = if (sessionPreferencesStore.isSendPublicReadReceiptsEnabled().first()) {
+                        ReceiptType.READ
+                    } else {
+                        ReceiptType.READ_PRIVATE
+                    }
+                    client.getRoom(event.roomId)?.markAsRead(receiptType)
```

Comment:
Should we extract this logic somewhere? Also, Why we send the receipt type here and send null in the MessagesPresenter?

Reply:
in the MessagesPresenter we want to clear the `manual` flag (just when opening the Room). The RR is eventually sent by the TimelinePresenter. In the case of a manual "Mark as read" we clear everything.